### PR TITLE
feat: add support for Next v13

### PR DESCRIPTION
**Pull request by dstaley**
🎟️ [Asana Task]()

---

## Description

This PR updates `@hashicorp/platform-nextjs-plugin` to use the latest version of `next-trasnpile-modules`, which is needed for Next v13 compatibility. It also disables `next-transpile-modules` if `nextConfig.experimental.transpileModules` is set.

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203348896548575

### CODE REVIEW

**Comment by dstaley** [comment]
I believe the test failures are related to [this bug](https://github.com/martpie/next-transpile-modules/pull/284) which should hopefully be fixed soon!

File: packages/nextjs-plugin/src/index.ts [hunk 1]
```diff
@@ -51,12 +51,22 @@ function withHashicorp({
...
 
-    const hcPackages = getHashicorpPackages(process.cwd())
+    if (
+      nextConfig.experimental &&
+      (nextConfig.experimental as { transpileModules?: string[] })
+        .transpileModules
+    ) {
+      debugLog(
+        'Disabling next-transpile-modules in favor of experimental transpileModules.'
+      )
```
Comment (BRKalow) [inline packages/nextjs-plugin/src/index.ts:61]: I think we'll still want to pass along the detected hashi packages to the new setting, right?

Comment (dstaley) [reply packages/nextjs-plugin/src/index.ts]: Good call! [`af95c5a` (#112)](https://github.com/hashicorp/web-platform-packages/pull/112/commits/af95c5a72b5001f388aab004f6d47c734ba46ed6)

**Comment by BRKalow** [comment]
@dstaley I'm inclined to merge this. I think we can either opt to require next v13 as a peer, or leave it as-is with your type adjustments (consumers should have the correct config type for `transpilePackages` if they are on v13). What do ya think?

**Comment by dstaley** [comment]
Yeah I'm good with merging! I think until we know the v13 upgrade is imminent for both dev portal and www that we should strive to ship versions that are both v12 and v13 compatible. (Once those are on v13 I think it's fine to start shipping stuff that's only compatible with v13)

**Review by BRKalow — APPROVED** [review]
Thanks for doing this! ❤️

**Pull request merged**